### PR TITLE
fix the bug of concurrent create snippet pipelines

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What type of this PR
/kind bugfix

#### What this PR does / why we need it:
fix the bug of concurrent create snippet pipelines

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/all?id=252139&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIl0sInN0YXRlcyI6WzQ0MDIsNzEwNCw3MTA1LDQ0MDMsNDQwNCw3MTA2LDQ0MDYsNDQwNyw0NDEyLDQ1MzgsNDQxMyw0NDE0LDQ0MTUsNDQxNl0sImFzc2lnbmVlSURzIjpbIjEwMDEyNjEiXX0%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=680&type=BUG)
- [Erda Cloud Issue Link](https://erda-org.erda.cloud/erda/dop/projects/387/issues/all?id=249137&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIl0sInN0YXRlcyI6WzQ0MDIsNzEwNCw3MTA1LDQ0MDMsNDQwNCw3MTA2LDQ0MDYsNDQwNyw0NDEyLDQ1MzgsNDQxMyw0NDE0LDQ0MTUsNDQxNl0sImFzc2lnbmVlSURzIjpbIjEwMDEyMDUiXX0%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=680&type=BUG)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |              |
| 🇨🇳 中文    |              |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/pipeline/services/pipelinesvc/snippet.go [hunk 1]
```diff
@@ -283,8 +283,12 @@ func (s *PipelineSvc) MakeSnippetPipeline4Create(p *spec.Pipeline, snippetTask *
 	for k, v := range snippetTask.Extra.Action.Params {
 		runParams = append(runParams, apistructs.PipelineRunParam{Name: k, Value: v})
 	}
+
+	labels := make(map[string]string)
```
Comment (sfwn) [inline modules/pipeline/services/pipelinesvc/snippet.go:287]: add comment for it.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/pipeline/services/pipelinesvc/snippet.go [hunk 2]
```diff
@@ -283,8 +283,12 @@ func (s *PipelineSvc) MakeSnippetPipeline4Create(p *spec.Pipeline, snippetTask *
 	for k, v := range snippetTask.Extra.Action.Params {
 		runParams = append(runParams, apistructs.PipelineRunParam{Name: k, Value: v})
 	}
+	// solve concurrent problem
+	labels := make(map[string]string)
```
Comment (sfwn) [inline modules/pipeline/services/pipelinesvc/snippet.go:287]: init size.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/pipeline/services/pipelinesvc/snippet.go (hunk 1 above)
Comment (littlejiancc) [reply modules/pipeline/services/pipelinesvc/snippet.go]: done

File: modules/pipeline/services/pipelinesvc/snippet.go (hunk 2 above)
Comment (littlejiancc) [reply modules/pipeline/services/pipelinesvc/snippet.go]: done

**Review by sfwn — APPROVED** [review]

**Pull request merged**